src/lib/netlist/devices/nld_4042.cpp

```diff
@@ -0,0 +1,100 @@
+// license:GPL-2.0+
+// copyright-holders:Couriersud, Jonathan Gevaryahu
+/*
+ * nld_4042.cpp
+ *
+ *  CD4042BM/CD4042BC Quad Clocked D Latch
+ *
+ *       +--------------+
+ *    Q4 |1     ++    16| VDD
+ *    Q1 |2           15| Q4Q
+ *   Q1Q |3           14| D4
+ *    D1 |4    4042   13| D3
+ *   CLK |5           12| Q3Q
+ *   POL |6           11| Q3
+ *    D2 |7           10| Q2
+ *   VSS |8            9| Q2Q
+ *       +--------------+
+ *
+ *
+ *          Function table
+ *
+ *          +-----+-----+----++----+-----+
+ *          | POL | CLK | Dx || Qx | QQx |
+ *          +=====+=====+====++====+=====+
+ *          |  0  |  0  |  X || Qx | /Qx |
+ *          |  0  |  1  |  D ||  D | /D  |
+ *          |  1  |  1  |  X || Qx | /Qx |
+ *          |  1  |  0  |  D ||  D | /D  |
+ *          +-----+-----+----++----+-----+
+ *          Note that since this is a level triggered transparent latch,
+ *          as long as POL ^ CLK == true, the latch is transparent, and
+ *          if D changes Q and QQ(/Q) will instantly change.
+ *
+ *  Naming conventions follow National Semiconductor datasheet
+ *
+ */
+
+#include "nld_4042.h"
+#include "nl_base.h"
+
+namespace netlist
+{
+namespace devices
+{
+
+
+	NETLIB_OBJECT(CD4042)
+	{
+		NETLIB_CONSTRUCTOR(CD4042)
```

Comment:
This is a CMOS device - use NETLIB_CONSTRUCTOR_MODEL(CD4029, "CD4XXX")

Reply:
Fixed.